### PR TITLE
Added DrawRectangle overload accepting RectangleF

### PR DESCRIPTION
**Pull request by medo64**
FillRectangle contains the following 4 overloads:
* public void FillRectangle(Brush brush, RectangleF rect)
* public void FillRectangle(Brush brush, float x, float y, float width, float height)
* public void FillRectangle(Brush brush, Rectangle rect)
* public void FillRectangle(Brush brush, int x, int y, int width, int height)

However, DrawRectangle, contains only the following 3 overloads:
* public void DrawRectangle(Pen pen, Rectangle rect)
* public void DrawRectangle(Pen pen, float x, float y, float width, float height)
* public void DrawRectangle(Pen pen, int x, int y, int width, int height)

In the interest of symmetry, I propose adding the "missing" RectangleF overload:
* public void DrawRectangle(Pen pen, RectangleF rect)

Since overload taking individual elements already exists, this new overload will just "decompose" rectangle structure into individual elements and call the existing function. Approach is exactly the one used in FillRectangle code.

### CODE REVIEW

**Comment by Wraith2** [comment]
If there is an approved api issue for this can you link to it please? If not you will need to go through the [Api Approval Process](https://github.com/dotnet/runtime/blob/main/docs/project/api-review-process.md ) before it is added.

**Comment by medo64** [comment]
> If there is an approved api issue for this can you link to it please? If not you will need to go through the [Api Approval Process](https://github.com/dotnet/runtime/blob/main/docs/project/api-review-process.md) before it is added.

I have added the [API proposal](https://github.com/dotnet/runtime/issues/62401). Will wait for result of the same. :)

**Comment by teo-tsirpanis** [comment]
Since you are adding a new API you also have to add it in the reference assembly's file. https://github.com/dotnet/runtime/blob/main/src/libraries/System.Drawing.Common/ref/System.Drawing.Common.cs#L508-L512

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=62385) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=62385) before we can accept your contribution.<br/><br/>:x: medo64 [sign now](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=62385)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/runtime?pullRequest=62385) it.</sub>

**Comment by medo64** [comment]
> Since you are adding a new API you also have to add it in the reference assembly's sources.

Added, thank you.

**Comment by safern** [comment]
@medo64 thanks for your contribution, given that the API Proposal issue has to be approved first, let's wait for it get reviewed by the review board and if approved we can reopen this PR. I will close the PR to follow the API Proposal guidelines on the repository, and we can reopen when ready.

**Comment by medo64** [comment]
[API proposal](https://github.com/dotnet/runtime/issues/62401) has been approved. Please reopen this pull request.

**Comment by safern** [comment]
@medo64 could you mind adding a test for each added API

**Comment by teo-tsirpanis** [comment]
@medo64 I made a mistake previously about the reference assembly's sources, you need to add the new API definitions to `src/libraries/System.Drawing.Common/ref/System.Drawing.Common.netcoreapp.cs` instead of `System.Drawing.Common.cs` (these new APIs will be available only for .NET Core). Please move them from the latter to the former.

**Comment by safern** [comment]
@teo-tsirpanis the other overloads are available for all target frameworks, so I think it is fine to keep them in all targets.

**Comment by teo-tsirpanis** [comment]
Well, I got confused, the .NET Framework 4.6.1 assembly on NuGet has only type forwards on the framework library, not actual types.

**Comment by medo64** [comment]
> @medo64 could you mind adding a test for each added API

For `DrawRectangle` I added an entry with `RectangleF` in every existing test I could find. Since other overloads are already actually tested here, I believe this is appropriate.

For `FillPie` there were no current test so I added basic argument tests modeled after tests present for `DrawPie`.

I do have two extra questions in regards to tests though:

1) I see that `Fill*` methods are missing even basic argument tests. What is the process for adding the same? Should I open a `[bug]` and wait for approval or should I just add them to this PR?

2) Test case named `DrawPie_ZeroHeight_ThrowsArgumentException` actually tests `DrawArc` method (my guess is that's copy/paste error). Should I just make this minor correction here or should I open a new `[bug]` and wait for approval?

File: src/libraries/System.Drawing.Common/src/System/Drawing/Graphics.cs
Comment (stephentoub) [inline src/libraries/System.Drawing.Common/src/System/Drawing/Graphics.cs:713]: Please also add XML comments for the parameters.

File: src/libraries/System.Drawing.Common/src/System/Drawing/Graphics.cs
Comment (stephentoub) [inline src/libraries/System.Drawing.Common/src/System/Drawing/Graphics.cs:1338]: Ditto

File: src/libraries/System.Drawing.Common/src/System/Drawing/Graphics.cs
Comment (medo64) [reply src/libraries/System.Drawing.Common/src/System/Drawing/Graphics.cs:713]: I can do that. I originally skipped it because none of other overloads has them.

File: src/libraries/System.Drawing.Common/src/System/Drawing/Graphics.cs
Comment (medo64) [reply src/libraries/System.Drawing.Common/src/System/Drawing/Graphics.cs:1340]: Added.

**Comment by medo64** [comment]
Anything else you fill needs to be added or this is it?

**Comment by safern** [comment]
I think @teo-tsirpanis was right and we need to move the APIs to the `ref/netcoreapp.cs`, that should fix the build. Also the tests should only be included for .NET Core targets, you can achieve that by adding: `[SkipOnTargetFramework(TargetFrameworks.NetFramework)]` attribute to the tests.

**Comment by medo64** [comment]
> you can do that by putting them on their own `.netcoreapp.cs` test file

I see that there are some tests already in GraphicsTests.Core.cs and that one seems to be targeting NETCoreApp. Should I add tests in this one or you want me to create a new file instead?

**Comment by medo64** [comment]
> need to move the APIs to the `ref/netcoreapp.cs`

Do you mean `System.Drawing.Common.netcoreapp.cs` in my case? I cannot find just plain `netcoreapp.cs` file...

**Comment by safern** [comment]
> I see that there are some tests already in GraphicsTests.Core.cs and that one seems to be targeting NETCoreApp. Should I add tests in this one or you want me to create a new file instead?

Adding them to that file seems better.

> Do you mean System.Drawing.Common.netcoreapp.cs in my case?

Yeah, I missed the `*` to state that.

**Review by safern — APPROVED** [review]

**Comment by safern** [comment]
Thank you @medo64!

**Pull request merged**
